addons/stock/models/stock_lot.py

```diff
@@ -165,6 +174,18 @@ def copy(self, default=None):
             default['name'] = _("(copy of) %s", self.name)
         return super().copy(default)
 
+    def _search_product_qty(self, operator, value):
+        if operator not in ('<', '>', '=', '!=', '<=', '>='):
+            raise UserError(_('Invalid domain operator %s', operator))
+        if not isinstance(value, (float, int)):
+            raise UserError(_('Invalid domain right operand %s', value))
+
+        ids = []
+        for lot in self.with_context(prefetch_fields=False).search([], order='id'):
+            if OPERATORS[operator](lot['product_qty'], value):
+                ids.append(lot.id)
```

Comment:
Searching for all lots in the database seems very bad for performance, instead we should search for `stock.quants` since thats how the field `product_qty` is computed. something like this should do the trick
```suggestion
        domain = ['|', ('location_id.usage', '=', 'internal'), '&', ('location_id.usage', '=', 'transit'), ('location_id.company_id', '!=', False)]
        quants_groupby = self.env['stock.quant']._read_group(domain, ['lot_id'], ['quantity:sum'])
        for lot, quantity_sum in quants_groupby:
            lot_id = lot.id
            if OPERATORS[operator](quantity_sum, value):
                ids.append(lot_id)
```